script/aria.js

```diff
@@ -764,16 +425,229 @@ function ariaAttributeReferences() {
+
+        placeholder.innerHTML = `All elements of the base markup except for the following roles: ${item.roles
+            .map((role) => `<rref>${role}</rref>`)
+            .join(", ")}`;
+        return;
     }
+
+    // Otherwise, i.e.,
+    // Cases: placeholderText "Placeholder" or "Use as a global deprecated in ARIA 1.2"
+
+    // populate placeholder
+    placeholder.innerHTML = `<ul>\n${item.roles
+        .map((role) => `<li><rref>${role}</rref></li>\n`)
+        .join("")}</ul>\n`;
+
+    // also update any inherited roles
+    const placeholderInheritedRoles = section.querySelector(
+        ".state-descendants, .property-descendants"
+    );
+    let inheritedRoles = new Set();
+    item.roles.forEach(function (role) {
+        // Some subroles have required properties which are also required by the superclass.
+        // Example: The checked state of radio, which is also required by superclass checkbox.
+        // We only want to include these one time, so filter out the subroles.
+        if (!descendantRoles[role]) return;
+        descendantRoles[role].forEach((subrole) => {
+            if (subrole.indexOf(propList[item.name].roles) === -1)
+                inheritedRoles.add(subrole);
+            // TODO: the if-check doesn't make sense
+            // Should it be the other way around? I.e.
+            // if (propList[item.name].roles.indexOf(subrole) === -1)
+            //     inheritedRoles.add(subrole);
+            // But this changes the spec, adding some, removing other entries
+        });
+    });
+
+    placeholderInheritedRoles.innerHTML = `<ul>\n${[...inheritedRoles]
+        .sort()
+        .map((role) => `<li><rref>${role}</rref></li>\n`)
+        .join("")}</ul>\n`;
+};
+
+/**
+ * In Object.entries loop, generates HTML for child role entries
+ * @param {String} role - subRoles key
+ * @param {Object} subRolesSet - subRoles value
+ */
+const generateHTMLRoleChildren = ([role, subroleSet]) => {
+    const item = [...subroleSet];
+    document.querySelector(`#${role} .role-children`).innerHTML = `<ul>\n${item
+        .map((subrole) => `<li><rref>${subrole}</rref></li>\n`)
+        .join("")}</ul>\n`;
+};
+
+function ariaAttributeReferences() {
+    const propList = {};
+    const globalSP = [];
+
+    let skipIndex = 0;
+    const myURL = document.URL;
+    if (myURL.match(/\?fast/)) {
+        skipIndex = 1;
+    }
+
+    // process the document before anything else is done
+    // first get the properties
+    const pdefsAndsdefs = document.querySelectorAll("pdef, sdef");
+    const pdefsAndsdefsContainer = [...pdefsAndsdefs].map(
+        (node) => node.parentNode
+    );
+
+    pdefsAndsdefs.forEach(populatePropList.bind(null, propList));
+    pdefsAndsdefs.forEach(populateGlobalSP.bind(null, propList, globalSP));
+    pdefsAndsdefs.forEach(generateHTMLStatesAndProperties.bind(null, propList));
```

Comment:
Each of these new function names could maybe follow a bit better of conventions... my top preference is at at least to rename any function that changes or produces html to have the prefix `render...`, and to describe specifically what it is creating -- like, `renderStatesAndPropertiesHeadings` instead of `generateHTMLStatesAndProperties`.

Also, related, I guess the "populate" is fine, but I might use a word like `build...` for anything that is just producing a data structure. Like `buildPropList`. 

Also.... I know you are reusing variable names, but personally I would use no short hands -- so, "buildStateAndPropertyList" instead of "buildPropList". and "buildGlobalStatesAndPropertiesList". I guess "pdefs" and "sdefs" etc are ok, considering they are using in our spec.

bu right now I still don't really understand why there is a `globalSP` and `propList` and the difference between these two things.. could they somehow be names a bit more clearly? Or are you trying to reuse these original names to make it easier to compare to the original script?